src/vs/workbench/parts/scm/electron-browser/scmViewlet.ts

```diff
@@ -1279,6 +1330,14 @@ export class SCMViewlet extends PanelViewlet implements IViewModel {
 		this.mainPanel.hide(repository);
 	}
 
+	shutdown(): void {
+		const [panel] = this.repositoryPanels;
+		if (panel) {
+			this.viewletSettings['query.commitMessagesHistory'] = panel.getCommitMessagesHistory();
```

Comment:
You're only saving this for one repository... what about all the others? We probably should scope this to the repository's provider ID and rootURI.